src/message_types.rs

```diff
@@ -0,0 +1,134 @@
+#![allow(dead_code)]
+
+use tokio::sync::mpsc;
+use tokio::sync::mpsc::error::SendError;
+use tokio::sync::mpsc::{Receiver, Sender};
+
+/// Represents a procedure call with transaction ID and command arguments
+pub struct Procedure {
+    xid: u32,
+    args: Command,
+}
+
+/// Enum of supported protocol commands
+pub enum Command {
+    NFSv3Command,
+    MountCommand,
+}
+
+/// Response container with transaction ID and procedure result
+pub struct Reply {
+    pub xid: u32,
+    pub result: ProcResult,
+}
+
+/// Result of procedure execution - either success or error
+pub enum ProcResult {
```

Comment:
Single file `proc.rs`. We should use rust namespaces (proc::Result) --- not C style names